source/common/network/address_impl.cc

```diff
@@ -126,21 +141,30 @@ Ipv4Instance::Ipv4Instance(const std::string& address, uint32_t port,
   }
 
   friendly_name_ = absl::StrCat(address, ":", port);
-  validateIpv4Supported(friendly_name_);
   ip_.friendly_address_ = address;
 }
 
 Ipv4Instance::Ipv4Instance(uint32_t port, const SocketInterface* sock_interface)
     : InstanceBase(Type::Ip, sockInterfaceOrDefault(sock_interface)) {
+  throwOnError(validateProtocolSupported());
   memset(&ip_.ipv4_.address_, 0, sizeof(ip_.ipv4_.address_));
   ip_.ipv4_.address_.sin_family = AF_INET;
   ip_.ipv4_.address_.sin_port = htons(port);
   ip_.ipv4_.address_.sin_addr.s_addr = INADDR_ANY;
   friendly_name_ = absl::StrCat("0.0.0.0:", port);
-  validateIpv4Supported(friendly_name_);
```

Comment:
Why can't `initHelper()` or some variant be used here?

Reply:
Friendly ping

Reply:
because initHelper() is used to remove duplicate code in throw and not throw version of ctors,  we don't have a not throw version of this ctor.